model: Prevent duplicate flags when processing update_message_flags.

sarafarajnasardi opened the pull request:
Previously, the event handler would unconditionally add flags to the message's flag list. If a flag (like 'read') was already present, this created duplicates, which broke subsequent removal operations.

This commit adds a check to ensure the flag is not already present before adding it, making the operation idempotent.

Fixes #1986

chrisbobbe commented on test/model/message_test.dart:
```diff
@@ -1702,6 +1702,22 @@ void main() {
+        // Should verify no notifications since no actual change happened
+        // But current implementation might notify anyway if it doesn't check for change
+        // optimization. The main point is flags list shouldn't have duplicates.
+        // Let's check the flags first.
```
I appreciate wanting to explain this 🙂 but I think we can cut this comment; the test is about an edge case that should be rare (in the fetch-event race) and an extra `notifyListeners` isn't a significant concern here.

chrisbobbe commented on lib/model/message.dart:
```diff
@@ -736,7 +736,9 @@ class MessageStoreImpl extends HasChannelStore with MessageStore, _OutboxMessage
-        message.flags.add(event.flag);
+        if (!message.flags.contains(event.flag)) {
+          message.flags.add(event.flag);
+        }
```
Please also write a test exercising this case.

chrisbobbe commented on test/model/message_test.dart:
```diff
@@ -1702,6 +1702,15 @@ void main() {
       });
+
+      test('avoid duplicate flags', () async {
+        await prepare();
```
```suggestion
      test('avoid duplicate flags', () async {
        // Regression test for https://github.com/zulip/zulip-flutter/issues/1986
        await prepare();
```

chrisbobbe replied:
Yes, I think so; it's applies equally there.

chrisbobbe reviewed (commented):
Thanks! Small comments below, otherwise LGTM.

gnprice reviewed (commented):
Thanks @sarafarajnasardi for taking care of this, and thanks @chrisbobbe for the previous reviews!

Generally this looks good. Some small comments, in addition to the one Chris makes above.

gnprice commented:
@sarafarajnasardi Do you think you'll have time to come back and finish addressing the feedback on this PR? It'd be great to have it completed.

sarafarajnasardi commented:
Hi @gnprice  and @chrisbobbe ,

Apologies for the delay in addressing the feedback. I encountered some personal matters that slowed my progress, and I accidentally closed the PR due to an incorrect force push while updating the changes.

I've now addressed all the review comments:

1.Fixed the duplicate flags issue for both all: true and all: false cases
2.Renamed variables to avoid abbreviations (message1, message2)
3.Used mkAddEvent helper and package:checks cascade shorthand
4.Updated commit message with proper prefix and formatting
The PR is now ready for review. Thank you for your patience and detailed feedback!
